src/clusterfuzz/_internal/google_cloud_utils/storage.py

```diff
@@ -249,23 +248,37 @@ def list_blobs(self, remote_path, recursive=True, names_only=False):
     else:
       fields = None
 
-    iterator = bucket.list_blobs(
-        prefix=path, delimiter=delimiter, fields=fields)
-    for blob in iterator:
-      properties['bucket'] = bucket_name
-      properties['name'] = blob.name
-      properties['updated'] = blob.updated
-      properties['size'] = blob.size
-
-      yield properties
-
-    if not recursive:
-      # When doing delimiter listings, the "directories" will be in `prefixes`.
-      for prefix in iterator.prefixes:
-        properties['bucket'] = bucket_name
-        properties['name'] = prefix
+    iterations = 0
+    while True:
```

Comment:
Is there no metadata in the http/grpc call that hints how many elements will be returned?

Reply:
I don't think so.